<html>
<!-- Mirrored from biostat.jhsph.edu/~iruczins/teaching/kogo/ by HTTrack Website Copier/3.x [XR&CO'2014], Fri, 28 Feb 2025 06:58:07 GMT -->
<!-- Added by HTTrack --><meta http-equiv="content-type" content="text/html;charset=UTF-8" /><!-- /Added by HTTrack -->
<head><title>Statistical Genomics</title>
<link rel=stylesheet type="text/css" href="ingo.css"></head>

<font size=+1><b>Statistical Genomics</b></font>
<br>
<font size=-2>Asian Institute in Statistical Genetics and Genomics, July 21-22, 2017.</font>

<hr>

<p><font color="red">Last update:</font> 
Thu Jul 20 04:39:03 EDT 2017

<hr>

<p><b>Outline </b> 

We will discuss a selection of high-throughput technologies to obtain genomic measurements, and the subsequent analyses. Emphasis will be on data processing to eliminate technological biases and to address unwanted variability, and on simple principles for proper and powerful Biostatistical inference. Discussion topics will include the role of genomic measurements for individualized and public health, experimental design, data pre-processing, dimension reduction, variance stabilization, statistical inference (including disease association), multiple comparisons, and enrichment analyses. We emphasize the importance of data provenance and reproducible research. Practical illustrations using data from DNA and RNA sequencing studies, genomic arrays, methylation and proteomics are given during the lectures. Additional software examples using <a href="http://www.bioconductor.org/">freely available Bioconductor tools</a> will be available on the course web page for students to practice.

<p>This course is intended for researchers who are actively engaged in genomics research and interested beginners, including laboratory scientists and clinicians with a basic quantitative background, and quantitatively trained researchers with basic knowledge in human genetics. The lectures will be given in English. Participants are expected to have some basic knowledge of core statistical principles (such as hypothesis testing and p-values). Basic knowledge of the <a href="http://cran.r-project.org/">open source statistical environment R</a> will be helpful to take full advantage of the examples given in class.

<hr>

<p><b>Additional material -</b>

Please see the material on the <a href="html/materials.html">additional material page</a> for more on relevant software, case studies, online classes, etc. 

<hr>

<p>
<b>Credits -</b> 
The slides and code contain material from classes for Biologists and Statisticians we teach at Johns Hopkins 
[ <a href="../140.615/index.html">140.615</a> ] 
[ <a href="http://www.biostat.jhsph.edu/~khansen/teaching/2014/140.668/">140.688</a> ]  
[ <a href="../140.751/index.html">140.751</a> ]  
[ <a href="../140.752/index.html">140.752</a> ]  
with material from
Karl Broman [ KB ], 
Kasper Hansen [ KH ], 
Rafa Irizarry [ RI ], 
Ben Langmead [ BL ], 
Jeff Leek [ JL ], 
Luigi Marcionni [ LM ], and
Rob Scharpf [ RS ]
which is gratefully acknowledged.

<hr>

<table cellspacing="10" cellpadding="3"><tbody>

<tr height="10"><td valign="top"></td></tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.introstatgenomics.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Introduction to statistical genomics</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.summarizing.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Summarizing and presenting genomic data</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.statmodel.twogroup.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Statistical modeling I : means and two-group comparisons</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.multiplecomp.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Multiple hypothesis testing</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.diffexpression.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Differential expression</td>
</tr>

<tr height="10"><td valign="top"></td></tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.pathway.genesets.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Pathway and gene set analyses</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.design.batch.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Experimental design</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.dimensionreduction.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Dimension reduction</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.batcheffects.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Batch effects</td>
</tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.statmodel.linearmodels.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Statistical modeling II : linear models in genomics</td>
</tr>

<tr height="10"><td valign="top"></td></tr>

<tr>
  <td valign="center" align="center">
  <a href="notes/n.statmodel.preprocessing.pdf"><img src="jpg/pdf.logo.jpg" border="0" width=20></a></td>
  <td valign="center">Statistical modeling III : pre-processing genomic data</td>
</tr>

</tbody></table>

<br><hr> 

</body>
<!-- Mirrored from biostat.jhsph.edu/~iruczins/teaching/kogo/ by HTTrack Website Copier/3.x [XR&CO'2014], Fri, 28 Feb 2025 06:58:07 GMT -->
</html>




